ui/src/data-products/data-products-container-with-provider.tsx

```diff
@@ -4,17 +4,42 @@ import React from 'react';
 import { PersistGate } from 'redux-persist/integration/react';
 import { persistStore } from 'redux-persist';
 import ElementListView from 'data-products/element-list/element-list-view';
+import VerticalGeometryView from 'data-products/vertical-geometry/vertical-geometry-view';
 
 const persistorDp = persistStore(dataProductsStore);
 
-const DataProductsMainContainer = connect()(ElementListView);
+type SelectedDataProduct = 'ELEMENT_LIST' | 'VERTICAL_GEOMETRY';
 
-export const DataProductsMainContainerWithProvider: React.FC = () => {
-    return (
-        <Provider store={dataProductsStore}>
-            <PersistGate loading={null} persistor={persistorDp}>
-                <DataProductsMainContainer />
-            </PersistGate>
-        </Provider>
-    );
+export type DataProductsContainerWithProviderProps = {
+    selectedDataProduct: SelectedDataProduct;
+};
```

Comment:
Mmh. Vaikka alunperin itse teinkin tämän `DataProductsContainerWithProvider`in juuri tällaiseen käyttöön, niin nyt kun tätä katson, niin mietin että voisi sittenkin olla fiksumpaa että elementtilistalla ja pystygeometrioilla olisikin omat `ContainerWithProvider`it, jotka sidottaisiin pathin perusteella tuolla `app-bar`issa. Tämä yhden Containerin ratkaisu näemmä johtaakin siihen, että täällä joudutaan iffittelemään kumpi tapaus on kyseessä , vaikka se menisi helpommin tuolla ylempänä jos nämä eriyttäisikin toisistaan

Reply:
Erotin kahdeksi eri provideriksi, ja se ratkaisu näyttää tosiaan siistimmältä